python/cugraph/graph/c_graph.pyx

```diff
@@ -452,3 +452,33 @@ class Graph:
         err = gdf_add_adj_list(g)
         cudf.bindings.cudf_cpp.check_gdf_error(err)
         return g.adjList.offsets.size - 1
+
+    def in_degree(self):
+        return self._degree(x=1)
+
+    def out_degree(self):
+        return self._degree(x=2)
+
+    def degree(self):
+        return self._degree()
+
+    def _degree(self, x = 0):
+        cdef uintptr_t graph = self.graph_ptr
+        cdef gdf_graph* g = < gdf_graph *> graph
+        err = gdf_add_adj_list(g)
```

Comment:
It is a bit extreme to compute the adj list here just to get the number of vertices.
First, we should check the transposed too.
If both are missing we could do n=max (max(src),max(dest)) which would be much faster

Ideally, when adj list and/or transposed adj list exist we should get the degree from the `offset` which would be much faster. Just need a simple kernel that does `deg[i] = offset[i+1]-offset[i]`

Reply:
If only edge list is present, adjacency list is calculated to get vertex identifiers and their corresponding degree. 
Added kernel for degree calculation from offsets